cfme/tests/intelligence/reports/test_generate_report.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+
+""" This test generate one default report for each category under reports accordion
+
+"""
+
+import pytest
+
+from cfme.infrastructure.provider.virtualcenter import VMwareProvider
+from cfme.intelligence.reports.reports import CannedSavedReport
+from utils import testgen
+from cfme import test_requirements
+
+pytestmark = [pytest.mark.tier(3),
+              test_requirements.report,
+              pytest.mark.usefixtures('setup_provider')]
+
+pytest_generate_tests = testgen.generate([VMwareProvider], scope='module')
+
+
+report_path = [
+    ["Configuration Management", "Virtual Machines", "Guest OS Information - any OS"],
+    ["Migration Readiness", "Virtual Machines", "Summary - VMs migration ready"],
+    ["Operations", "Virtual Machines", "VMs not Powered On"],
+    ["VM Sprawl", "Candidates", "Summary of VM Create and Deletes"],
+    ["Relationships", "Virtual Machines, Folders, Clusters", "VM Relationships"],
+    ["Events", "Operations", "Events for VM prod_webserver"],
+    ["Performance by Asset Type", "Virtual Machines", "Top CPU Consumers (weekly)"],
+    ["Running Processes", "Virtual Machines", "Processes for prod VMs sort by CPU Time"],
+    ["Trending", "Clusters", "Cluster CPU Trends (last week)"],
+    ["Tenants", "Tenant Quotas", "Tenant Quotas"],
+    ["Provisioning", "Activity Reports", "Provisioning Activity - by VM"],
+]
+
+
+@pytest.mark.parametrize('path', report_path, scope="module")
+def test_reports_generate_report(request, path):
+    """ This Tests run one default report for each category
+
+    Steps:
+        *Run one default report
+        *Delete this Saved Report from the Database
+    """
+    report = CannedSavedReport.new(path)
```

Comment:
REQUIRED: if report creation is failed, that report won't be deleted. I'd recommend to use something from the following:
1. request.addfinalizer()
2.  yield fixture finalization 
3. try: finally: